component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -203,17 +207,30 @@ protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
         fileNameList = fileNameList.replaceAll("\\s", "");
         fileObjectList = Arrays.asList(fileNameList.split(","));
         for (int i = 0; i < fileObjectList.size(); i++) {
-            String fileObjectPath = listeningDirUri + "/" + fileObjectList.get(i);
+            String fileObjectPath = tmpURL + File.separator + fileObjectList.get(i);
             listeningFileObject = Utils.getFileObject(fileObjectPath);
+            String filePath = null;
             try {
+                filePath = listeningFileObject.getURL().toURI().toURL().getPath();
+                if (!filePath.contains(File.separator)) {
```

Comment:
Shouldnt we write this as`if(filePath.contains("/"))`

Reply:
When we get the value from listeningFileObject it is always gonna treat it as linux or Unix system and create the file path with "/" but I guess its better to go as u suggested . Will do

Reply:
I think we need it like !filePath.contains(File.separator) because File.seperator gives the value depending on the OS and we need only to run this if path contains "/" and but actual value need to be "\" wdyt ?